torchbenchmark/models/pyhpc_isoneutral_mixing/isoneutral_pytorch.py

```diff
@@ -0,0 +1,213 @@
+                dSdz[2:-2, 1 + jp:-2 + jp, :su]
+            syn = -drodyn / (torch.min(torch.tensor([0.], device=device), drodzn) - epsln)
+            taper = dm_taper(syn)
+            sumz[:, :, ki:] += dzw[None, None, :su] \
+                * maskV[2:-2, 1:-2, ki:] * torch.max(torch.tensor([K_iso_steep], device=device), diffloc[2:-2, 1:-2, ki:] * taper)
+            Ai_nz[2:-2, 1:-2, ki:, jp, kr] = taper * \
+                syn * maskV[2:-2, 1:-2, ki:]
+    K_22[2:-2, 1:-2, :] = sumz / (4. * dzt[None, None, :])
+
+    """
+    compute Ai_bx, Ai_by and K33 on top face of T cell.
+    """
+    sumx = torch.zeros_like(K_11)[2:-2, 2:-2, :-1]
+    sumy = torch.zeros_like(K_11)[2:-2, 2:-2, :-1]
+
+    for kr in range(2):
+        if kr == 1:
+            sl = 1
+            su = K_11.shape[2]
+        else:
+            sl = 0
+            su = K_11.shape[2] - 1
+
+        drodzb = drdT[2:-2, 2:-2, sl:su] * dTdz[2:-2, 2:-2, :-1] \
+            + drdS[2:-2, 2:-2, sl:su] * dSdz[2:-2, 2:-2, :-1]
+
+        # eastward slopes at the top of T cells
+        for ip in range(2):
+            drodxb = drdT[2:-2, 2:-2, sl:su] * dTdx[1 + ip:-3 + ip, 2:-2, sl:su] \
+                + drdS[2:-2, 2:-2, sl:su] * \
+                dSdx[1 + ip:-3 + ip, 2:-2, sl:su]
+            sxb = -drodxb / (torch.min(torch.tensor([0.], device=device), drodzb) - epsln)
+            taper = dm_taper(sxb)
+            sumx += dxu[1 + ip:-3 + ip, None, None] * \
+                K_iso[2:-2, 2:-2, :-1] * taper * \
+                sxb**2 * maskW[2:-2, 2:-2, :-1]
+            Ai_bx[2:-2, 2:-2, :-1, ip, kr] = taper * \
+                sxb * maskW[2:-2, 2:-2, :-1]
+
+        # northward slopes at the top of T cells
+        for jp in range(2):
+            facty = cosu[1 + jp:-3 + jp] * dyu[1 + jp:-3 + jp]
+            drodyb = drdT[2:-2, 2:-2, sl:su] * dTdy[2:-2, 1 + jp:-3 + jp, sl:su] \
+                + drdS[2:-2, 2:-2, sl:su] * \
+                dSdy[2:-2, 1 + jp:-3 + jp, sl:su]
+            syb = -drodyb / (torch.min(torch.tensor([0.], device=device), drodzb) - epsln)
+            taper = dm_taper(syb)
+            sumy += facty[None, :, None] * K_iso[2:-2, 2:-2, :-1] \
+                * taper * syb**2 * maskW[2:-2, 2:-2, :-1]
+            Ai_by[2:-2, 2:-2, :-1, jp, kr] = taper * \
+                syb * maskW[2:-2, 2:-2, :-1]
+
+    K_33[2:-2, 2:-2, :-1] = sumx / (4 * dxt[2:-2, None, None]) + \
+        sumy / (4 * dyt[None, 2:-2, None]
+                * cost[None, 2:-2, None])
+    K_33[2:-2, 2:-2, -1] = 0.
+
+    return K_11, K_22, K_33, Ai_ez, Ai_nz, Ai_bx, Ai_by
+
+
+def prepare_inputs(*inputs, device):
+    out = [torch.as_tensor(a, device='cuda' if device == 'gpu' else 'cpu') for a in inputs]
+    if device == 'gpu':
+        torch.cuda.synchronize()
+    return out
+
+
+def run(*inputs, device='cpu'):
+    with torch.no_grad():
+        if device == 'gpu':
+            outputs = isoneutral_diffusion_pre_gpu(*inputs)
+            torch.cuda.synchronize()
```

Comment:
hmm this is good but we also do it again in the benchmark wrapper so i'm not sure if it adds extra overhead we care about.

Reply:
I don't think this path is actually run, so I'll delete it (and `prepare_inputs` as above)